### PR TITLE
`--alias` fix

### LINKED ISSUE #532: confusing behavior when passing  `--alias`
when running `%sql CONNECTION_STRING --alias ALIAS`, a new connection is not created if there's an active connection with duckdb connection string, but this should not be the case, a new connection should be created:

```
In [1]: %load_ext sql

In [2]: %sql --connections
Out[2]:
Active connections:
+---------+-----+-------+
| current | url | alias |
+---------+-----+-------+
+---------+-----+-------+

In [3]: %sql duckdb://

In [4]: %sql --connections
Out[4]:
Active connections:
+---------+-----------+-------+
| current |    url    | alias |
+---------+-----------+-------+
|    *    | duckdb:// |       |
+---------+-----------+-------+

In [5]: %sql duckdb:// --alias newconnection

In [6]: %sql --connections
Out[6]:
Active connections:
+---------+-----------+-------+
| current |    url    | alias |
+---------+-----------+-------+
|    *    | duckdb:// |       |
+---------+-----------+-------+
```

### PR DESCRIPTION
**Pull request by AnirudhVIyer**
## Describe your changes
Modified the set method in the Connection Class such that: if a connection with a descriptor exists, we check it's alias with the new alias argument.
## Issue number

Closes #532 

## Checklist before requesting a review

- [x] Performed a self-review of my code
- [x] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [x] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [x] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

----
:books: Documentation preview :books:: https://jupysql--600.org.readthedocs.build/en/600/

### CODE REVIEW

**Review by edublancas — CHANGES_REQUESTED** [review]
don't worry about the formatting changes (unsure about why it's happening but we can merge it)

just add some unit tests

File: src/sql/connection.py [hunk 1]
```diff
@@ -369,20 +368,27 @@ def set(cls, descriptor, displaycon, connect_args=None, creator=None, alias=None
...
                 # is that we're missing some unit tests
                 # when descriptor is a connection object
                 # http://docs.sqlalchemy.org/en/rel_0_9/core/engines.html#custom-dbapi-connect-arguments # noqa
-                cls.current = existing or Connection.from_connect_str(
-                    connect_str=descriptor,
-                    connect_args=connect_args,
-                    creator=creator,
-                    alias=alias,
-                )
+                # if same alias found
```
Comment (mehtamohit013) [inline src/sql/connection.py:378]: @edublancas 
Please check whether this is a desired behavior. If yes, the logic looks good, and it is working on binder

**Review by mehtamohit013 — CHANGES_REQUESTED** [review]

File: src/tests/test_connection.py [hunk 2]
```diff
@@ -377,3 +377,16 @@ def test_close_all(ip_empty):
         connections_copy["duckdb://"].execute("").fetchall()
 
     assert not Connection.connections
+
+
+def test_new_connection_with_alias(ip_empty):
```
[the remark this answers is not in the document]
Comment (AnirudhVIyer) [reply src/tests/test_connection.py:382]: Sure ! Will add them

**Comment by anupam-tiwari** [comment]
@AnirudhVIyer I think you removed your code

**Comment by AnirudhVIyer** [comment]
> @AnirudhVIyer I think you removed your code

Yeah, an integration test is failing, trying to figure that out

**Comment by mehtamohit013** [comment]
@AnirudhVIyer 
It seems like other PR also have test failing. Make sure you merge with the master and test again

**Comment by edublancas** [comment]
@tonykploomber: can you check the integration tests? looks like they're failing but this PR didn't change them. so perhaps some update in the docker image we're using?

**Comment by jinniw43805** [comment]
@edublancas 
This is something we found earlier, https://github.com/ploomber/jupysql/issues/614

**Review by edublancas — APPROVED** [review]

**Comment by edublancas** [comment]
great work!

**Comment by jinniw43805** [comment]
@edublancas 
Any idea why [tests are skipped](https://github.com/ploomber/jupysql/actions/runs/5263047607/jobs/9512827286) when merged to master?

**Comment by mehtamohit013** [comment]
> @edublancas
> Any idea why [tests are skipped](https://github.com/ploomber/jupysql/actions/runs/5263047607/jobs/9512827286) when merged to master?

It may be due to my recently merged [PR](https://github.com/ploomber/jupysql/pull/598). In `check_doc` job, I am taking the `git diff` of the current branch with the `master` branch. This works great when running actions for a PR, but when merging, there is no `diff` between the `master` latest commit and the current branch i.e. `master` latest commit. May have to modify the test to run only for PR, and run all the tests when merging 

What is your opinion on the above issue? @edublancas @jinniw43805 @tonykploomber

**Pull request merged**